packages/react-components/src/components/CameraButton.tsx

```diff
@@ -139,44 +148,62 @@ export const CameraButton = (props: CameraButtonProps): JSX.Element => {
   const [waitForCamera, setWaitForCamera] = useState(false);
   const localeStrings = useLocale().strings.cameraButton;
   const strings = { ...localeStrings, ...props.strings };
-  const onRenderCameraOnIcon = (): JSX.Element => (
-    <HighContrastAwareIcon disabled={props.disabled || waitForCamera} iconName="ControlButtonCameraOn" />
-  );
-  const onRenderCameraOffIcon = (): JSX.Element => (
-    <HighContrastAwareIcon disabled={props.disabled || waitForCamera} iconName="ControlButtonCameraOff" />
-  );
+  const [announcerString, setAnnouncerString] = useState<string>(strings.cameraActionTurnedOffAnnouncement);
+  const onRenderCameraOnIcon = (): JSX.Element => {
+    return <HighContrastAwareIcon disabled={props.disabled || waitForCamera} iconName="ControlButtonCameraOn" />;
+  };
+  const onRenderCameraOffIcon = (): JSX.Element => {
+    return <HighContrastAwareIcon disabled={props.disabled || waitForCamera} iconName="ControlButtonCameraOff" />;
+  };
   if (waitForCamera && strings.tooltipVideoLoadingContent) {
     strings.tooltipDisabledContent = strings.tooltipVideoLoadingContent;
   }
 
   const onToggleClick = useCallback(async () => {
     // Throttle click on camera, need to await onToggleCamera then allow another click
     if (onToggleCamera) {
+      // allows for the setting of narrator strings triggering the announcer when camera is turned on or off.
+      if (announcerString === strings.cameraActionTurnedOffAnnouncement) {
+        setAnnouncerString(strings.cameraActionTurnedOnAnnouncement);
+      } else {
+        setAnnouncerString(strings.cameraActionTurnedOffAnnouncement);
+      }
       setWaitForCamera(true);
       try {
         await onToggleCamera(localVideoViewOptions ?? defaultLocalVideoViewOptions);
```

Comment:
Move the `toggleAnnouncerString` to after `await onToggleCamera` as it should be announced after the action has completed successfully.
e.g. if this fails to turn on the video it shouldn't have announced that the video was turned on.